sdk/identity/identity/review/identity.api.md

```diff
@@ -139,6 +139,14 @@ export class InteractiveBrowserCredential implements TokenCredential {
 
 // @public
 export interface InteractiveBrowserCredentialOptions extends TokenCredentialOptions {
+    // Warning: (ae-forgotten-export) The symbol "AuthenticationRecord" needs to be exported by the entry point index.d.ts
+    authenticationRecord?: AuthenticationRecord;
+    cacheOptions?: {
+        cachePlugin?: {
+            readFromStorage: () => Promise<string>;
+            writeToStorage: (getMergedState: (oldState: string) => string) => Promise<void>;
```

Comment:
feels kinda bad that write isn't the same shape as read... what does getMergedState() do?

Reply:
This comes from having to merge our changes into the cache rather than just writing out (other applications and/or other languages may have written to the shared cache)

Reply:
Right, so you always hand back `oldState` being the last string you read in with `readFromStorage` and then write that? Or something else?

Reply:
Generally, the user won't even fulfill this contract. We'll soon be providing helper functions that will create the storage/persistence plugin for you.

I'm tempted to chalk this up as some of the underlying implementation leaking through a little bit, though I'm not sure how to resolve as we still currently allow users to implement their own persistence plugins.

@schaabs - any thoughts?

Reply:
We've been back and forth on whether we'll be adding APIs to allow people to write the cache themselves. If we can hide these for now it would be ideal. But given it's a preview if we can't easily hide these in the short term we could include them, but we should be clear it's subject to change.